src/module_wrap.cc

```diff
@@ -650,13 +673,9 @@ Maybe<URL> Resolve(Environment* env,
   }
   if (ShouldBeTreatedAsRelativeOrAbsolutePath(specifier)) {
     URL resolved(specifier, base);
-    Maybe<URL> file = ResolveExtensions<TRY_EXACT_NAME>(resolved);
-    if (!file.IsNothing())
-      return file;
-    if (specifier.back() != '/') {
-      resolved = URL(specifier + "/", base);
-    }
-    return ResolveDirectory(env, resolved, check_pjson_main);
+    if (ResolvesToFile(resolved))
+      return Just(resolved);
```

Comment:
Beautiful :)

Reply:
That's all @zenparsing

Reply:
BTW, this C++ code is a pleasure to work with 👍

Reply:
All the implementation work of @bmeck, with many various refactoring contribs.